### PR TITLE
Аdd sdk total liquidity calculation

### PR DESCRIPTION
**Pull request by alter-eggo**
# Description
Аdd sdk total liquidity calculation for single pool.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have requested at least 2 reviews (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/services/pool/decorators/pool.decorator.ts [hunk 1]
```diff
@@ -39,18 +38,20 @@ export class PoolDecorator {
     const processedPools = this.pools.map(pool => {
       const poolService = new this.poolServiceClass(pool);
       poolService.removeBptFromTokens();
-      poolService.setTotalLiquidity(prices, currency, tokens);
       poolService.setUnwrappedTokens();
       return poolService.pool;
     });
 
     const poolMulticaller = new PoolMulticaller(processedPools);
+    const { pools: sdkPools } = balancer;
```
Comment (garethfuller) [inline src/services/pool/decorators/pool.decorator.ts:46]: How do you know the pools have been fetched here?

File: src/services/pool/decorators/pool.decorator.ts [hunk 2]
```diff
@@ -69,6 +70,11 @@ export class PoolDecorator {
       poolService.setFeesSnapshot(poolSnapshot);
       poolService.setVolumeSnapshot(poolSnapshot);
       await poolService.setLinearPools();
+      if (sdkPool) {
+        pool.totalLiquidity = await sdkPools.liquidity(sdkPool);
```
Comment (garethfuller) [inline src/services/pool/decorators/pool.decorator.ts:74]: As Jarek said, the SDK function can handle a pool even if the schema is not the exact same. You might be able to just do:
```suggestion
        pool.totalLiquidity = await sdkPools.liquidity(poolService.pool);
```

File: src/services/pool/decorators/pool.decorator.ts [hunk 3]
```diff
@@ -69,6 +70,11 @@ export class PoolDecorator {
...
+      } else {
+        poolService.setTotalLiquidity(prices, currency, tokens);
```
Comment (garethfuller) [inline src/services/pool/decorators/pool.decorator.ts:76]: I'd prefer we don't use this calculation at all anymore, so if the SDK calculation fails we should fallback to the subgraph value. That might even be what is happening in the SDK function too.

**Review by garethfuller — CHANGES_REQUESTED** [review]

File: src/services/pool/decorators/pool.decorator.ts (hunk 1 above)
Comment (garethfuller) [reply src/services/pool/decorators/pool.decorator.ts:46]: Sorry, forget this comment, I mistook what was going on here.

File: src/services/pool/decorators/pool.decorator.ts (hunk 3 above)
Comment (alter-eggo) [reply src/services/pool/decorators/pool.decorator.ts]: removed

**Review by garethfuller — APPROVED** [review]

**Pull request merged**